snap/validate_test.go

```diff
@@ -149,6 +149,294 @@ func (s *ValidateSuite) TestValidateAppName(c *C) {
 	}
 }
 
+func (s *ValidateSuite) TestValidateAppSockets(c *C) {
+	socket := &SocketInfo{
+		Name:         "sock",
+		ListenStream: "/var/snap/mysnap/common/socket",
+		SocketMode:   0600,
+	}
+	app := AppInfo{
+		Snap: &Info{
+			SideInfo: SideInfo{
+				RealName: "mysnap",
+				Revision: Revision{20},
+			},
+		},
+		Name:  "foo",
+		Plugs: map[string]*PlugInfo{"network-bind": {}},
+		Sockets: map[string]*SocketInfo{
+			"sock": socket,
+		},
+	}
```

Comment:
Can we please have this logic in a function that returns the app, and then used it on every function below?  Each function can tune the good base as necessary for its own needs.

Reply:
definitely nicer, done